encoding: Fix low bitrate check

Penwy opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

<!--- Make sure you’ve read the contribution guidelines here: https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst -->

### Description
<!--- Describe your changes in detail. -->
<!--- If this change includes UI elements, please include screenshots. -->
This fixes the "Low Stream Bitrate" check to work with the new encoder names, and reworks it to work with all of obs' default encoders, and be less disruptible.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open GitHub Issue, or implements feature request -->
<!--- from the Ideas page, please link to the issue here. -->
The "Low Stream Bitrate" check is currently nonfunctional, due to the encoders rename in OBS 28.

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment (hardware, OS version, etc.),-->
<!--- and the tests you ran, including how it may affect other areas of code. -->
This was tested by manually feeding the lines to the function and checking it gets the right values.
It was tested with
- x264
- "new" nvenc and ffmpeg nvenc
- qsv11
- texture amf and fallback amf

### Types of changes
<!--- What types of changes does your PR introduce? Uncomment all that apply -->
<!--- - Bug fix (non-breaking change which fixes an issue) -->
<!--- - New feature (non-breaking change which adds functionality) -->
<!--- - Tweak (non-breaking change to improve existing functionality) -->
<!--- - Performance enhancement (non-breaking change which improves efficiency) -->
<!--- - Code cleanup (non-breaking change which makes code smaller or more readable) -->
<!--- - Breaking change (fix or feature that would cause existing functionality to change) -->
<!--- - Documentation (a change to documentation pages) -->

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

RytoEX commented:
Do you have logs to test this against?

Penwy commented:
here's for :
- [x264](https://obsproject.com/logs/0OIN1QXHbhM2iDxx)
- [obs-nvenc](https://obsproject.com/logs/efolcmOvAbVAV0c5)
- [fallback qsv](https://obsproject.com/logs/nOhEeUqhsjOAjGZx)
- [new obs-nvenc (31.x)](https://obsproject.com/logs/49Esc0sWSG5FMgyb)

Those are the encoders I have easy access to right now, if you want more I can find, but it'll take a bit more time.
(took the time to fix a few oversights I found while testing those).

Fenrirthviti reviewed (approved):
Changes look fine to me conceptually. No reason to have a specific check per-encoder for this. The intent might have been at some point to expand this check to do other things, but we're probably well past that.

RytoEX commented:
> here's for :
> 
>     * [x264](https://obsproject.com/logs/0OIN1QXHbhM2iDxx)
> 
>     * [obs-nvenc](https://obsproject.com/logs/efolcmOvAbVAV0c5)
> 
>     * [fallback qsv](https://obsproject.com/logs/nOhEeUqhsjOAjGZx)
> 
>     * [new obs-nvenc (31.x)](https://obsproject.com/logs/49Esc0sWSG5FMgyb)
> 

The [fallback qsv](https://obsproject.com/logs/nOhEeUqhsjOAjGZx) log causes an error with this PR:
```
Traceback (most recent call last):
  File "C:\dev\obsproject\loganalyzer\loganalyzer.py", line 220, in <module>
    main()
  File "C:\dev\obsproject\loganalyzer\loganalyzer.py", line 214, in main
    msgs = doAnalysis(url=flags.url, filename=flags.file)
  File "C:\dev\obsproject\loganalyzer\loganalyzer.py", line 166, in doAnalysis
    checkStreamSettings(logLines),
  File "C:\dev\obsproject\loganalyzer\checks\encoding.py", line 67, in checkStreamSettings
    match = params_re.search(lines[line])
IndexError: list index out of range
```

Penwy commented:
fixed the regex, and added a try except so that, if another unforeseen issue happens with the regexp it fails the check gracefully instead of stopping the whole analysis.

RytoEX commented on checks/encoding.py:
```diff
@@ -3,6 +3,9 @@
 import re
 
 
+params_re = re.compile(r"\t(?P<key>\w+):?\s*(?P<value>\S+)")
```
Is there any known case where the colon is missing/optional? `:?`

RytoEX replied:
I think we can make it non-optional.
